adapter: fix off-by-one in connection id generation

mjibson opened the pull request:
We were 1 bit high on the max connection id which ended up conflicting with the mask.

Fixes #24574

### Motivation

  * This PR fixes a recognized bug.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  <!-- Reference the design in the description. -->
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  <!-- Ask in #team-cloud on Slack if you need help preparing the cloud PR. -->
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - n/a

jkosh44 commented on src/ore/src/id_gen.rs:
```diff
@@ -299,12 +302,51 @@ mod internal {
+/// Number of bits the org id is offset into a connection id.
+pub const ORG_ID_OFFSET: usize = 19;
+
+/// Number of bits the org id is offset into a connection id.
```
Is this comment accurate?

mjibson replied:
Fixed.

mjibson commented:
I realized that we already had a supposed check for this, but it also had an off by one error. Fixed that as well and added another test that would have found this.
